### PR TITLE
Recreate imagestream.yaml removed from upstream and add PR check for it

### PR DESCRIPTION
**Pull request by rgdoliveira**
Many thanks for submitting your Pull Request :heart:! 

Please make sure your PR meets the following requirements:

- [ ] You have read the [contributors guide](README.md#contributing-to-kogito-images-repository)
- [ ] Pull Request title is properly formatted: `[KOGITO|RHPAM-XYZ] Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a testcase that verifies it
- [ ] You've tested the new feature/bug fix in an actual OpenShift cluster
- [ ] You've added a [RELEASE_NOTES.md](RELEASE_NOTES.md) entry regarding this change

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- <b>(Re)run Jenkins tests</b>  
  Please add comment: <b>Jenkins [test|retest] this</b>

- <b>Prod tests</b>  
  Please add comment: <b>Jenkins (re)run [prod|Prod|PROD]</b>
</details>

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

**Review by mareknovotny — APPROVED** [review]

File: openshift-serverless-logic-imagestream.yaml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+kind: ImageStreamList
+apiVersion: v1
+metadata:
+  name: openshift-serverless-kogito-image-streams
```
Comment (ricardozanini) [inline openshift-serverless-logic-imagestream.yaml:22]:
```suggestion
  name: openshift-serverless-logic-image-streams
```

File: openshift-serverless-logic-imagestream.yaml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  annotations:
+    description: ImageStream definitions for Red Hat OpenShift Serverless Logic images
+    openshift.io/provider-display-name: Kie Group
+items:
+  - kind: ImageStream
+    apiVersion: v1
+    metadata:
+      name: logic-data-index-ephemeral-rhel8
+      annotations:
+        openshift.io/display-name: Red Hat build of Runtime image for Kogito Data Index Service for ephemeral PostgreSQL persistence provider
```
Comment (ricardozanini) [inline openshift-serverless-logic-imagestream.yaml:32]:
```suggestion
        openshift.io/display-name: Red Hat build of Runtime image for SonataFlow Data Index Service for ephemeral PostgreSQL persistence provider
```

File: openshift-serverless-logic-imagestream.yaml [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        openshift.io/provider-display-name: Kie Group
+    spec:
+      tags:
+      - name: '1.32.0'
+        annotations:
+          description: Red Hat build of Kogito Data Index Service - ephemeral PostgreSQL
```
Comment (ricardozanini) [inline openshift-serverless-logic-imagestream.yaml:38]:
```suggestion
          description: Red Hat build of SonataFlow Data Index Service - ephemeral PostgreSQL
```

File: openshift-serverless-logic-imagestream.yaml [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+          iconClass: icon-jbpm
+          tags: logic-data-index,kogito,data-index,data-index-ephemeral
```
Comment (ricardozanini) [inline openshift-serverless-logic-imagestream.yaml:40]:
```suggestion
          tags: logic-data-index,sonataflow,data-index,data-index-ephemeral
```

File: openshift-serverless-logic-imagestream.yaml [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+          type: Local
+        from:
+          kind: DockerImage
+          name: registry.redhat.io/openshift-serverless-1-tech-preview/logic-data-index-ephemeral-rhel8:1.32.0
+  - kind: ImageStream
+    apiVersion: v1
+    metadata:
+      name: logic-devmode
+      annotations:
+        openshift.io/display-name: Red Hat build of Kogito Serverless Workflow development mode image with Quarkus extensions libraries preinstalled.
```
Comment (ricardozanini) [inline openshift-serverless-logic-imagestream.yaml:53]:
```suggestion
        openshift.io/display-name: Red Hat build of SonataFlow development mode image with Quarkus extensions libraries preinstalled.
```

File: openshift-serverless-logic-imagestream.yaml [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+        openshift.io/provider-display-name: Kie Group
+    spec:
+      tags:
+        - name: '1.32.0'
+          annotations:
+            description: Red Hat build of Kogito Serverless Workflow development mode image with Quarkus extensions libraries preinstalled.
```
Comment (ricardozanini) [inline openshift-serverless-logic-imagestream.yaml:59]:
```suggestion
            description: Red Hat build of SonataFlow development mode image with Quarkus extensions libraries preinstalled.
```

File: openshift-serverless-logic-imagestream.yaml [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+            iconClass: icon-jbpm
+            tags: logic,devmode,kogito,kogito,development,serverless,workflow
```
Comment (ricardozanini) [inline openshift-serverless-logic-imagestream.yaml:61]:
```suggestion
            tags: logic,devmode,sonataflow,development,serverless,workflow
```

File: openshift-serverless-logic-imagestream.yaml [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+            type: Local
+          from:
+            kind: DockerImage
+            name: registry.redhat.io/openshift-serverless-1-tech-preview/logic-swf-devmode-rhel8:1.32.0
+  - kind: ImageStream
+    apiVersion: v1
+    metadata:
+      name: logic-builder
+      annotations:
+        openshift.io/display-name: Red Hat build of Runtime image for Kogito Serverless Workflow builder with Quarkus extensions libraries preinstalled
```
Comment (ricardozanini) [inline openshift-serverless-logic-imagestream.yaml:74]:
```suggestion
        openshift.io/display-name: Red Hat build of Runtime image for SonataFlow builder with Quarkus extensions libraries preinstalled
```

File: openshift-serverless-logic-imagestream.yaml [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+        openshift.io/provider-display-name: Kie Group
+    spec:
+      tags:
+        - name: '1.32.0'
+          annotations:
+            description: Red Hat build of Runtime image for Kogito Serverless Workflow builder with Quarkus extensions libraries preinstalled
```
Comment (ricardozanini) [inline openshift-serverless-logic-imagestream.yaml:80]:
```suggestion
            description: Red Hat build of Runtime image for SonataFlow builder with Quarkus extensions libraries preinstalled
```

File: openshift-serverless-logic-imagestream.yaml [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+            iconClass: icon-jbpm
+            tags: logic,builder,kogito,workflow,serverless
```
Comment (ricardozanini) [inline openshift-serverless-logic-imagestream.yaml:82]:
```suggestion
            tags: logic,builder,sonataflow,workflow,serverless
```

**Review by ricardozanini — COMMENTED** [review]
Let's use this opportunity to remove Kogito naming from our bits!

**Comment by rgdoliveira** [comment]
@ricardozanini thanks, I've applied your suggestions

**Review by ricardozanini — APPROVED** [review]

**Comment by ricardozanini** [comment]
@rgdoliveira mind taking a look at the bats test that's failing?

**Comment by rgdoliveira** [comment]
Let's discuss it on the other PR (from Fabrizio) and merge this one.

**Pull request merged**